### DISCUSSION TITLE
please support nuttx os

### DISCUSSION THREAD

**Discussion by snikeguo** [dotnet/runtime#105123] [General] [answered]
home page:
https://nuttx.apache.org/
pthread:
https://nuttx.apache.org/docs/latest/reference/user/08_pthread.html

**Comment by am11** [comment, marked as answer]
Porting .NET to new platform is a non-trivial work. You can see the ongoing port work:

FreeBSD: https://github.com/dotnet/runtime/issues/14537
illumos: https://github.com/dotnet/runtime/issues/34944
Haiku: https://github.com/dotnet/runtime/issues/55803
QNX: https://github.com/dotnet/runtime/issues/33374

Someone familiar with this platform will need to step up and do the ground work first, get the port to the point where lowest abstractions are working. Then ask for help. It shouldn't be impossible, but the good start would be to provide a development environment with system dependencies installed (or preferably cross-build setup with that we can use with llvm, see https://github.com/dotnet/arcade/blob/main/eng/common/cross/build-rootfs.sh).

**Comment by ladeak** [comment]
I think [WildernessLabs](https://github.com/WildernessLabs/) is using nuttx for their meadow platform.

**Reply by am11** [reply]
Nice. They seem to have built a layer in their OS on top of NuttX to support .NET runtime. Then they extended the BCL (written in C#) which interops with nuttx and exposed functionality to managed code.

**Reply by snikeguo** [reply]
@am11   the WildernessLabs is using Interpreter  mode . not jit or aot mode

**Comment by shaofing** [comment]
https://mp.weixin.qq.com/s/_UdrZZbv3Zs_pA6wmmED0g 
已经支持了
already supported
